compute_endpoint/globus_compute_endpoint/cli.py

```diff
@@ -717,6 +817,85 @@ def self_diagnostic(compress: bool, log_kb: int):
         click.echo(f"Successfully created {filename}")
 
 
+def create_or_choose_auth_project(ac: AuthClient) -> str:
+    """
+    Lets the user choose from one of their existing projects, or
+    prompt for a description and creates one for them
+    """
+    projects = ac.get_projects()
+    if any(projects):
+        proj_selected = user_input_select(
+            (
+                "Please select a project to contain new auth policy "
+                "(leave blank to create a new one):"
+            ),
+            [f'id: {proj["id"]}, name: {proj["display_name"]}' for proj in projects],
+        )
+        if proj_selected:
+            return proj_selected.split()[1][:-1]
+
+    try:
+        print("Creating a new auth project.")
+
+        display_name = input("Enter a name for this project: ")
+        contact_email = input("Enter the contact email for this project: ")
+
+        admin_ids_input = input(
+            "Enter a comma separated list of admin IDs for this project: "
+        )
+        admin_ids = admin_ids_input.split(",") if admin_ids_input else []
+
+        admin_gps_input = input(
+            "Enter a comma separated list of group IDs as project admins: "
+        )
+        admin_group_ids = admin_gps_input.split(",") if admin_gps_input else []
```

Comment:
I must admit that I like this.  Prompting for information == UX I like.  Heh.  And I like that you haven't `isatty()` protected it, allowing pipeline and automated interaction with this code.

Reply:
Perhaps consider an "are you sure?" step just before actually making the `create_policy()` request, that shows the parsed information.  That gives the human a chance to make sure that they didn't typo something and that we parsed it correctly.